modules/component-store/src/component-store.ts

```diff
@@ -149,58 +155,71 @@ export class ComponentStore<T extends object> implements OnDestroy {
    *
    * @param projector A pure projection function that takes the current state and
    *   returns some new slice/projection of that state.
+   * @param config SelectConfig that changes the behavoir of selector, including
+   *   the debouncing of the values until the state is settled.
    * @return An observable of the projector results.
    */
-  select<R>(projector: (s: T) => R): Observable<R>;
-  select<R, S1>(s1: Observable<S1>, projector: (s1: S1) => R): Observable<R>;
+  select<R>(projector: (s: T) => R, config?: SelectConfig): Observable<R>;
+  select<R, S1>(
+    s1: Observable<S1>,
+    projector: (s1: S1) => R,
+    config?: SelectConfig
+  ): Observable<R>;
   select<R, S1, S2>(
     s1: Observable<S1>,
     s2: Observable<S2>,
-    projector: (s1: S1, s2: S2) => R
+    projector: (s1: S1, s2: S2) => R,
+    config?: SelectConfig
   ): Observable<R>;
   select<R, S1, S2, S3>(
     s1: Observable<S1>,
     s2: Observable<S2>,
     s3: Observable<S3>,
-    projector: (s1: S1, s2: S2, s3: S3) => R
+    projector: (s1: S1, s2: S2, s3: S3) => R,
+    config?: SelectConfig
   ): Observable<R>;
   select<R, S1, S2, S3, S4>(
     s1: Observable<S1>,
     s2: Observable<S2>,
     s3: Observable<S3>,
     s4: Observable<S4>,
-    projector: (s1: S1, s2: S2, s3: S3, s4: S4) => R
+    projector: (s1: S1, s2: S2, s3: S3, s4: S4) => R,
+    config?: SelectConfig
   ): Observable<R>;
-  select<R>(...args: any[]): Observable<R> {
-    let observable$: Observable<R>;
-    // project is always the last argument, so `pop` it from args.
-    const projector: (...args: any[]) => R = args.pop();
-    if (args.length === 0) {
-      // If projector was the only argument then we'll use map operator.
-      observable$ = this.stateSubject$.pipe(debounceSync(), map(projector));
+  select<
+    O extends Array<Observable<unknown> | SelectConfig | ProjectorFn>,
+    R,
+    ProjectorFn = (...a: unknown[]) => R
+  >(...args: O): Observable<R> {
+    const { observables, projector, config } = processSelectorArgs(args);
+
+    let observable$: Observable<unknown>;
+    // If there are no Observables to combine, then we'll just map the value.
+    if (observables.length === 0) {
+      observable$ = this.stateSubject$.pipe(
+        config.debounce ? debounceSync() : (source$) => source$,
+        map(projector)
+      );
     } else {
-      // If there are multiple arguments, we're chaining selectors, so we need
+      // If there are multiple arguments, then we're aggregating selectors, so we need
       // to take the combineLatest of them before calling the map function.
-      observable$ = combineLatest(args).pipe(
-        // The most performant way to combine Observables avoiding unnecessary
-        // emissions and projector calls.
-        debounceSync(),
-        map((args: any[]) => projector(...args))
+      observable$ = combineLatest(observables).pipe(
+        config.debounce ? debounceSync() : (source$) => source$,
```

Comment:
What if we would extract this ternary statement?
Not sure if it's worth it tho, I'm thinking that when this line changes, the line in the `if` case would also be changed.

Reply:
Yeah, I couldn't think of a better way to extract it and make it readable/somewhat easier to understand at the same time.